src/test/java/com/github/neatlife/jframework/util/PrecisionUtilTest.java

```diff
@@ -0,0 +1,137 @@
+package com.github.neatlife.jframework.util;
+
+import org.junit.Assert;
+import org.junit.Test;
+import java.math.BigDecimal;
+
+public class PrecisionUtilTest {
+
+    @Test
+    public void testBigDecimalAdd() {
+        Assert.assertEquals(BigDecimal.ONE,
+                PrecisionUtil.add(null, BigDecimal.ONE));
+        Assert.assertEquals(BigDecimal.ONE,
+                PrecisionUtil.add(BigDecimal.ONE, null));
+    }
+
+    @Test
+    public void testBigDecimalSubtract() {
+        Assert.assertEquals(new BigDecimal("-10"),
+                PrecisionUtil.subtract(null, BigDecimal.TEN));
+        Assert.assertEquals(BigDecimal.ONE,
+                PrecisionUtil.subtract(BigDecimal.ONE, null));
+    }
+
+    @Test
+    public void testBigDecimalMultiply() {
+        Assert.assertEquals(new BigDecimal("10"),
+                PrecisionUtil.multiply(null, BigDecimal.TEN));
+        Assert.assertEquals(BigDecimal.ONE,
+                PrecisionUtil.multiply(BigDecimal.ONE, null));
+    }
+
+    @Test
+    public void testBigDecimalDivide() {
+        Assert.assertEquals(new BigDecimal("0.10"),
+                PrecisionUtil.divide(BigDecimal.ONE, BigDecimal.TEN));
+        Assert.assertEquals(new BigDecimal("0.10"),
+                PrecisionUtil.divide(BigDecimal.ONE, BigDecimal.TEN));
+        Assert.assertEquals(new BigDecimal("0"),
+                PrecisionUtil.divide(null, BigDecimal.TEN, 1));
+        Assert.assertEquals(new BigDecimal("1.0"),
+                PrecisionUtil.divide(BigDecimal.ONE, null, 1));
+    }
+
+    @Test
+    public void testStringAdd() {
+        Assert.assertEquals("1", PrecisionUtil.add("1", null));
+        Assert.assertEquals("2", PrecisionUtil.add(null, "2"));
+    }
+
+    @Test
+    public void testStringSubtract() {
+        Assert.assertEquals("1", PrecisionUtil.subtract("1", null));
+        Assert.assertEquals("-2", PrecisionUtil.subtract(null, "2"));
+    }
+
+    @Test
+    public void testStringMultiply() {
+        Assert.assertEquals("1", PrecisionUtil.multiply("1", null));
+        Assert.assertEquals("2", PrecisionUtil.multiply(null, "2"));
+    }
+
+    @Test
+    public void testStringDivide() {
+        Assert.assertEquals("1.00", PrecisionUtil.divide("1", null));
+        Assert.assertEquals("0", PrecisionUtil.divide(null, "2"));
```

Comment:
Add a test in each of these four `testStringX` tests that does not have either parameter `null`.

Reply:
Done